src/Bindings/PluginLua.cpp

```diff
@@ -72,12 +72,13 @@ void cPluginLua::Close(void)
 	cResettablePtrs resettables;
 	std::swap(m_Resettables, resettables);
 	{
-		cCSUnlock Unlock(Lock);
+		m_CriticalSection.Unlock();
 		for (auto resettable: resettables)
 		{
 			resettable->Reset();
 		}
 		m_Resettables.clear();
+		m_CriticalSection.Lock();
```

Comment:
Also, the way the servers set up at the moment, if an exception is triggered, the server is about to stop, so there's no point unlocking a lock that's about to be destroyed.